src/Symfony/Component/Validator/Constraints/AbstractComposite.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+
+/**
+ * @Annotation
+ *
+ * @author Marc Morales Valldepérez <marcmorales83@gmail.com>
+ * @author Marc Morera Merino <yuhu@mmoreram.com>
+ */
+abstract class AbstractComposite extends Composite
+{
+    /**
+     * @var array
+     *
+     * Set of constraints
+     */
+    public $constraints = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct($options = null)
+    {
+        parent::__construct($options);
+
+        $this->constraints = (array) $this->constraints;
+
+        // Each constraint contained
+        foreach ($this->constraints as $constraint) {
+            if (!$constraint instanceof Constraint) {
+                throw new ConstraintDefinitionException(sprintf('The value %s is not an instance of Constraint in constraint %s', $constraint, __CLASS__));
```

Comment:
Should we use `get_class($this)` instead of `__CLASS__` here and everywhere else to include the class name of the concrete constraint instead of `AbstractComposite`?

Reply:
yes